### ISSUE TITLE
[Feature Request] Repository Specific External Editors

### ISSUE THREAD

**Issue by DylanDevelops** [desktop/desktop#18161]
### The feature request

I would love to be able to assign an external editor specifically to any specific repository that would override the main default external editor but only for the chosen repository.

I work a lot in Visual Studio Code, which is why it is my chosen external editor; however, sometimes I work on a Java project such as a Minecraft mod, for which I use IntelliJ IDEA or other types of projects requiring different IDEs. In cases like this, I would love to open my project easily in IDEA (as an example) with the GitHub Desktop shortcut while also being able to open other repositories in something else.

**TL;DR:** It would be amazing for specific repositories to be able to set up a different external editor to open the repository.

### Proposed solution

It would be nice if the specific editor were shown here or had two options: 1) The default external editor and 2) The repository-specific external editor.

<img width="583" alt="Screenshot 2024-02-08 at 10 51 54 PM" src="https://github.com/desktop/desktop/assets/48571264/d05d0d23-1348-4dc6-8509-a6ada0347d17">

**Comment by steveward** [comment]
Thanks for the issue @DylanDevelops. This is being tracked in https://github.com/desktop/desktop/issues/12195 so you can follow along there for updates. I'll add this feedback there as well.
